plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java

```diff
@@ -290,6 +290,29 @@ public void testDropSchemaExternalFiles()
 
     protected abstract String bucketUrl();
 
+    @Test
+    void testCreateDropDynamicCatalog()
+    {
+        String catalog = "new_catalog_" + randomNameSuffix();
+        String createCatalogSql = """
+                CREATE CATALOG %1$s USING delta_lake
+                WITH (
+                    "hive.metastore" = 'thrift',
+                    "hive.metastore.uri" = '%2$s'
+                )""".formatted(catalog, hiveHadoop.getHiveMetastoreEndpoint().toString());
+        assertUpdate(createCatalogSql);
+        assertCatalogs("system", "delta", "hive", "tpch", catalog);
+
+        assertUpdate("DROP CATALOG " + catalog);
+        assertCatalogs("system", "delta", "hive", "tpch");
+        // re-add the same catalog
+        assertUpdate(createCatalogSql);
+        assertCatalogs("system", "delta", "hive", "tpch", catalog);
```

Comment:
Is it going to be useful to issue a query since that would show any issues with "leftover state"? Or not really and it gets into resiliency/scale testing territory?

Reply:
ya it gets into resiliency, but we can maybe do a show schemas or something?